build.assets/Dockerfile-multiarch-base

```diff
@@ -0,0 +1,127 @@
+#
+# 64-bit centos7 provides devtools-10 for arm64, devtools-11 for amd64 and 
+# gcc 4.8 for 32-bit archs. This image builds modern gcc from sources. 
+# It guarantees that we would use the same GCC version for all builds, 
+# and we would be able to upgrade GCC to a newer version if required.
+# #############################################################################
+
+FROM centos:7 AS base
+
+# Automatically supplied by the Docker buildkit
+ARG TARGETARCH
+
+ARG GCC_VERSION=10.4.0
+ARG CMAKE3_VERSION=3.12.3
+
+# #############################################################################
+# Platform-specific customisation.
+# #############################################################################
+
+## ARM 32 #####################################################################
+FROM base AS platform-setup-arm
+ENV GCC_BUILD_FLAGS="--build arm-unknown-linux-gnueabi --with-float=hard --with-mode=arm"
+ENV GOLANG_ARCH=armv6l
+ENV RUST_ARCH=armv7-unknown-linux-gnueabihf
+
+ARG YUM_UPDATE_FLAGS=""
+RUN echo "armhfp" > /etc/yum/vars/basearch && \
+    echo "armv7hl" > /etc/yum/vars/arch && \
+    echo "armv7hl-redhat-linux-gnu" > /etc/rpm/platform
+
+## ARM 64 #####################################################################
+FROM base AS platform-setup-arm64
+ENV GCC_BUILD_FLAGS="--build aarch64-unknown-linux-gnu"
+ENV GOLANG_ARCH=arm64
+ENV RUST_ARCH=aarch64-unknown-linux-gnu
+
+# Installing the kerenel packages causes the update to hang on aarch64, so we
+# skip upgrading them.
+ARG YUM_UPDATE_FLAGS="--exclude kernel-*"
+RUN echo "aarch64-redhat-linux-gnu" > /etc/rpm/platform
+
+## 386 ########################################################################
+FROM base AS platform-setup-386
+ENV GCC_BUILD_FLAGS="--build i386-unknown-linux-gnu"
+ENV GOLANG_ARCH=386
+ENV RUST_ARCH=i686-unknown-linux-gnu
+
+## AMD 64 #####################################################################
+FROM base AS platform-setup-amd64
+ENV GCC_BUILD_FLAGS="--build x86_64-unknown-linux-gnu"
+ENV GOLANG_ARCH=amd64
+ENV RUST_ARCH=x86_64-unknown-linux-gnu
+
+## GCC built from sources #####################################################
+FROM platform-setup-$TARGETARCH AS gcc
+
+ENV LANGUAGE=en_US.UTF-8 \
+    LANG=en_US.UTF-8 \
+    LC_ALL=en_US.UTF-8 \
+    LC_CTYPE=en_US.UTF-8
+
+RUN yum update -y $YUM_UPDATE_FLAGS && \
+    yum install -y     \
+        git \
+        gcc \
+        gcc-c++ \
+        make \
+        pam-devel \
+        perl-IPC-Cmd \
+        tree \
+        which \
+        zip \
+        zlib-static \
+        automake \
+        libtool \
+        flex \        
+        autoconf-archive \
+        libudev-devel \
+        make \
+        wget \
+        gcc \
+        gcc-c++ \
+        glibc-static \
+        glibc-devel \
+        libstdc++-static \
+        libstdc++-devel \
+        bzip2 \
+        file \
+        binutils \
+        zlib-devel \
+        elfutils-libelf-devel-static \
```

Comment:
I don't think that `elfutils-libelf-devel-static` is needed here

Reply:
2023-03-02T10:55:04.2019152Z #31 1.247 Package libelf was not found in the pkg-config search path.
2023-03-02T10:55:04.2020851Z #31 1.247 Perhaps you should add the directory containing `libelf.pc'
2023-03-02T10:55:04.2021739Z #31 1.247 to the PKG_CONFIG_PATH environment variable

Reply:
Where is this log come from? AFAIK `libelf` is only needed by `libbpf`.

Reply:
From libbpf build log. I can install it separately at the libbpf build stage, but I guess it is better to keep all deps in one place.